### PR TITLE
feature: response_body_filters support two-dimensional array

### PR DESCRIPTION
**Pull request by detailyang**
this feature was mentioned on the google groups  [link](https://groups.google.com/forum/#!topic/openresty/bz5sORDVw7c)

### CODE REVIEW

**Comment by detailyang** [comment]
@agentzh  can you help me review this PR, many thanks:)

**Comment by detailyang** [comment]
@moonbingbing can you try this PR?

It has existed for a long time,  maybe you can review this PR together:)

**Comment by detailyang** [comment]
@agentzh  

Please help me review again, I have changed the wrong point :grin:

By the way, so sorry for long time to changes because of ignoring the github email :(.

**Comment by agentzh** [comment]
@detailyang Thanks for your update!

@doujiang24 Will you have a quick look at this PR too? ;)

File: lib/Test/Nginx/Socket.pm [hunk 1]
```diff
@@ -1230,64 +1270,66 @@ sub fmt_str ($) {
...
         }
+
+        if (ref $filter && ref $filter eq 'ARRAY') {
+
+            for my $f (@$filter) {
+                $new = run_filter_helper($block, $f, $new);
+            }
+
+        } else {
+
```
Comment (doujiang24) [inline lib/Test/Nginx/Socket.pm]: style: better remove this blank line ?

**Comment by doujiang24** [comment]
@detailyang Nice job :)
@agentzh  LGTM :)

**Comment by agentzh** [comment]
Merged. Thanks!

**Pull request closed without merging**